scripts/bugzilla_attach.py

```diff
@@ -0,0 +1,222 @@
+#!/bin/python3
+
+"""
+Author: Avram Levitter
+
+Parameters:
+    
+    id: The Bugzilla bug ID to send the attachment to
+    image: The image to use (defaults to quay.io/kubevirt/must-gather)
+    api-key: Use a generated API key instead of a username and login
+
+Requirements:
+
+    Openshift 4.1
+    Python 3.6
+    A Bugzilla account for www.bugzilla.redhat.com
+
+This script attaches the result of the must-gather command, as executed
+by the kubevirt must-gather image, to the supplied bugzilla id on the
+Red Hat bugzilla website.
+It first creates an output subdirectory in the working directory named
+gather-files/ and then executes the following command:
+'oc adm must-gather --image=quay.io/kubevirt/must-gather
+--dest-dir=gather-files/' and pipes the output to
+gather-files/must-gather.log
+In order to meet the maximum attachment file sizes, it searches the
+created directory for any log files that exceed 100,000 lines, and if
+there are any, it trims those files down to 100,000 lines.
+It then creates a time-stamped archive file to compress the attachment
+and prepare it for upload.
+After doing so, the attachment is encoded as a base64 string.
+In order to authenticate against the Bugzilla website, a username and
+password are prompted. A POST request is then sent to the Bugzilla
+website. If there are any errors (invalid ID or invalid login), the
+script prompts for those and retries the request until there are no
+errors.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+import tarfile
+import datetime
+import base64
+from getpass import getpass
+import requests
+
+MAX_LOGLINES = 100000
```

Comment:
How did you come up with this number?

Reply:
I'm trying to keep the compressed size of the files down to around 4MB, and 100,000 lines gives around that (give or take based on the file)

Reply:
How difficult would it be to base it on time? We could parse the time in log and get only last 5 mins (configurable).